tool/planet/etcd.go

```diff
@@ -437,6 +625,32 @@ func etcdRestore(file string) error {
 	return nil
 }
 
+func waitEtcdHealthyTimeout(ctx context.Context, timeout time.Duration, client etcd.Client) error {
+	ctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+
+	return trace.Wrap(waitEtcdHealthy(ctx, client))
+}
+
+// waitEtcdHealthy waits for etcd to have a leader elected
+func waitEtcdHealthy(ctx context.Context, client etcd.Client) error {
+	mapi := etcd.NewMembersAPI(client)
+
+	ticker := time.NewTicker(1 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return trace.Wrap(ctx.Err())
+		case <-ticker.C:
+			leader, _ := mapi.Leader(ctx)
```

Comment:
If the ignored argument here is an error, shouldn't we at least log it?

Reply:
Yea, I didn't want to log in the loop, because a number of errors are expected there. So I've updated it to capture the error, and include the error when returning due to timeout.